src/topic/__tests__/TopicContainer.test.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { configure, shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import configureMockStore from 'redux-mock-store';
+import 'jest-styled-components';
+import TopicContainer from '../TopicContainer';
+import thunk from 'redux-thunk';
+import TopicModel from '../../shared/models/TopicModel';
+import TopicsList from '../components/TopicsList';
+
+const mockedActionType = 'TopicContainerAction';
+
+jest.mock('../../actions/topic.actions', () => ({
+    getTopics: () => ({ type: mockedActionType })
+}));
+
+configure({ adapter: new Adapter() });
+const mockStore = configureMockStore([thunk]);
+
+describe('<TopicContainer>', () => {
+    const initialState = {
+        auth: {
+            loggedUser: false
+        },
+        topic: {
+            pending: true,
+            error: 'test error message',
+            topics: [TopicModel.fromBackendData({})]
+        }
+    };
+    let wrapper;
+    let store = mockStore(initialState);
+    
+    beforeEach(() => {
+        store.clearActions();
+        /**
+         * help would be appreciated. we have to (?) dive, otherwise we will not have the real actions binded to
+         * Component.
+         * 1) Passing it as props is a fake test (i.e we don't know if the real one is called
+         * 2) while using a dive, we have actions called immediately which may not be what we want
+         *
+         * I picked 2) as using 1) would only tell us that prop is called but we would lose the assurance that
+         * Container is correctly 'connected' (i.e. has correctly mapped dispatchToProps
```

Comment:
interesting, have you experienced some bugs with wrongly connected components before?

Reply:
@Tuhaj lets say so - lack of correct mapping or lack of `connect` at all :)